[arm] change prior box implement

chenjiaoAngel opened the pull request:
当输入shape不改变时，prior box的输出直接copy上次结果；否则重新计算输出
原因：prior box的输出结果与输入shape和参数相关

chenjiaoAngel committed:
Merge pull request #89 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
Merge pull request #90 from PaddlePaddle/develop

[operator] add InferShapeImpl method (#3294)

chenjiaoAngel committed:
add some op infershape implement, test=develop

chenjiaoAngel committed:
add reshape infershape, test=develop

chenjiaoAngel committed:
Merge pull request #91 from PaddlePaddle/develop

update code

chenjiaoAngel committed:
fix format， test=develop

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
fix space format. test=develop

chenjiaoAngel committed:
add conv_transpose+bn fusion. test=develop

chenjiaoAngel committed:
delete note, test=develop

chenjiaoAngel committed:
Merge pull request #92 from PaddlePaddle/develop

update code

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
update code. test=develop
Merge branch 'opencl' of https://github.com/chenjiaoAngel/Paddle-Lite into opencl

chenjiaoAngel committed:
fix format space, test=develop

chenjiaoAngel committed:
fix opt run error, test=develop

chenjiaoAngel committed:
add boxcoder opencl kernel, test=develop

chenjiaoAngel committed:
Merge pull request #93 from PaddlePaddle/develop

update code

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
add cmake, test=develop

chenjiaoAngel committed:
fix format. test=develop

chenjiaoAngel committed:
fix format. test=develop

chenjiaoAngel committed:
fix format aa. test=develop

chenjiaoAngel committed:
fix , test=develop

chenjiaoAngel committed:
Merge pull request #94 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
update profile info(add new element), test=develop

chenjiaoAngel committed:
Merge pull request #95 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
Merge pull request #96 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
Merge pull request #97 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
fix clang ut build error

chenjiaoAngel committed:
add gemm+relu6

chenjiaoAngel committed:
fix build error

chenjiaoAngel committed:
fix .h

chenjiaoAngel committed:
fix gemm_s8

chenjiaoAngel committed:
Merge pull request #98 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
fix ut conv+leakyRelu

chenjiaoAngel committed:
improve 3x3s1 direct profile

chenjiaoAngel committed:
update code
Merge branch 'opencl' of https://github.com/chenjiaoAngel/Paddle-Lite into opencl

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
pull code
Merge branch 'opencl' of https://github.com/chenjiaoAngel/Paddle-Lite into int8

chenjiaoAngel committed:
Merge pull request #107 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
update cide
Merge branch 'int8' of https://github.com/chenjiaoAngel/Paddle-Lite into int8

chenjiaoAngel committed:
add gemv+relu6/lleakyRelu

chenjiaoAngel committed:
fix v7 build bug

chenjiaoAngel committed:
fix relu6 bug

chenjiaoAngel committed:
fix gemm ut bug

chenjiaoAngel committed:
fix ut

chenjiaoAngel committed:
fix ut

chenjiaoAngel committed:
fi format. test=develop

chenjiaoAngel committed:
fix format. test=develop

chenjiaoAngel committed:
fic format. test=develop

chenjiaoAngel committed:
ff. test=develop

chenjiaoAngel committed:
fix v7 clang build error, test=develop

chenjiaoAngel committed:
fix v7 build register error, test=develop

chenjiaoAngel committed:
fix format.  test=develop

chenjiaoAngel committed:
fix build register error, ttest=develop

chenjiaoAngel committed:
fix build register error, ttest=develop

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
ff format,test=develop

chenjiaoAngel committed:
fix relu6 problem, test=develop

chenjiaoAngel committed:
fix form, test=develop

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
ff, test=develop

chenjiaoAngel committed:
add six / scale , test=develop

chenjiaoAngel committed:
fix conflicct

chenjiaoAngel committed:
iMerge branch 'PaddlePaddle-develop' into int8

fix conflict:wq
wq

chenjiaoAngel committed:
fix pooling overflow, test=develop

chenjiaoAngel committed:
fix conflict test=develop

chenjiaoAngel committed:
Merge pull request #116 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
pull code

chenjiaoAngel committed:
Merge pull request #120 from PaddlePaddle/develop

pull

chenjiaoAngel committed:
Merge pull request #121 from PaddlePaddle/develop

git pull code

chenjiaoAngel committed:
add grouup_norm

chenjiaoAngel committed:
pull code
Merge branch 'int8' of https://github.com/chenjiaoAngel/Paddle-Lite into int8

chenjiaoAngel committed:
fix format. test=develop

chenjiaoAngel committed:
fix foormat, test=develop

chenjiaoAngel committed:
fix format. test=develop

chenjiaoAngel committed:
fix ff.test=develoop

chenjiaoAngel committed:
fix xiaodu crash. test=develop

chenjiaoAngel committed:
format. test=develop

chenjiaoAngel committed:
Merge pull request #125 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
fix concatt axis < 0 errorr,ttest=develop

chenjiaoAngel committed:
fix format. test=develop

chenjiaoAngel committed:
Merge pull request #127 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
fix conv int8 kernel choose and sooftmax compute bug

chenjiaoAngel committed:
change axis_size = 4 kernel choose, test=develop

chenjiaoAngel committed:
fix format. test=develop

chenjiaoAngel committed:
Merge pull request #129 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
Merge pull request #131 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
Merge pull request #132 from PaddlePaddle/develop

pull new code

chenjiaoAngel committed:
Merge pull request #133 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
uupdate sequence_pool and sequence_conv profiler, test=develop

chenjiaoAngel committed:
fix format, testt=develop

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
fix format test=develop

chenjiaoAngel committed:
fix compute error. test=develop

chenjiaoAngel committed:
fix compute error

chenjiaoAngel committed:
fix compute error, test=develop

chenjiaoAngel committed:
pull
Merge branch 'int8' of https://github.com/chenjiaoAngel/Paddle-Lite into int8

chenjiaoAngel committed:
delete warning and extra info, test=develop

chenjiaoAngel committed:
Merge pull request #135 from PaddlePaddle/develop

pull code

chenjiaoAngel committed:
update sequence_conv profile

chenjiaoAngel committed:
add conv+conv(1x1s1p0) fusion

chenjiaoAngel committed:
fix build

chenjiaoAngel committed:
fix build

chenjiaoAngel committed:
fix ruun error

chenjiaoAngel committed:
fix conflict

chenjiaoAngel committed:
opt sucess

chenjiaoAngel committed:
add note

chenjiaoAngel committed:
fix run error, test=develop

chenjiaoAngel committed:
remove note test=develop

chenjiaoAngel committed:
fix conflict

chenjiaoAngel committed:
test=develop

chenjiaoAngel committed:
fix format

chenjiaoAngel committed:
fix conflict, test=develop

chenjiaoAngel committed:
fix conflict, test=develop

chenjiaoAngel committed:
fix formmat. test=develop

chenjiaoAngel committed:
fix formmat. test=develop

chenjiaoAngel committed:
fix formmat. test=develop

chenjiaoAngel committed:
test=develop

chenjiaoAngel committed:
pull
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle-Lite into PaddlePaddle-develop

chenjiaoAngel committed:
fix conflict

chenjiaoAngel committed:
update priorbox profile

chenjiaoAngel committed:
fix format. test=develop

chenjiaoAngel committed:
test=develop

chenjiaoAngel committed:
fix prior, test=develop

chenjiaoAngel committed:
pull test=develop
Merge branch 'fusion' of https://github.com/chenjiaoAngel/Paddle-Lite into fusion

hong19860320 commented on lite/kernels/arm/prior_box_compute.cc:
```diff
@@ -45,10 +46,318 @@ inline void ExpandAspectRatios(const std::vector<float>& input_aspect_ratior,
 
-void PriorBoxCompute::Run() {
-  auto& param = Param<operators::PriorBoxParam>();
+inline void* fast_malloc(size_t size) {
```
https://github.com/PaddlePaddle/Paddle-Lite/blob/develop/lite/backends/host/target_wrapper.cc 已经实现了，是否可以重用？

chenjiaoAngel replied:
这个可以的，我稍后改过来

chenjiaoAngel committed:
fix review. test=develop

chenjiaoAngel committed:
pulpulll
Merge branch 'fusion' of https://github.com/chenjiaoAngel/Paddle-Lite into fusion

chenjiaoAngel committed:
test=develop

hong19860320 commented on lite/kernels/arm/prior_box_compute.cc:
```diff
@@ -46,9 +48,301 @@ inline void ExpandAspectRatios(const std::vector<float>& input_aspect_ratior,
-void PriorBoxCompute::Run() {
-  auto& param = Param<operators::PriorBoxParam>();
+inline void fast_free(void* ptr) {
+  if (ptr) {
```
这个也可以删除哈

hong19860320 reviewed (approved):
LGTM
